src/main/java/software/amazon/cloudwatchlogs/emf/Constants.java

```diff
@@ -23,6 +23,8 @@ public class Constants {
 
     public static final int MAX_METRICS_PER_EVENT = 100;
 
+    public static final int MAX_DIMENSIONS = 30;
```

Comment:
nit: Dimensions is pretty broad in EMF (i.e. dimension set, dimension set entry, etc). What this means is actually the limit of dimensions inside a dimension set.. so maybe we can be a bit more specific on this name as well?
Something like `MAX_DIMENSION_SET_SIZE`